# Type: feat
# Subtype: E2
# English Summary: Removed empty projects from template list
# Chinese Summary: 从模板列表中删除了空项目
# Entry Number: 8
# Beomsu : fix, cleaning up unwanted items, improving existing behavior

# === Git Diff Content ===

diff --git a/java/idea-ui/src/com/intellij/platform/templates/PlainModuleTemplatesFactory.java b/java/idea-ui/src/com/intellij/platform/templates/PlainModuleTemplatesFactory.java
index 2457b3b..fb8d72e 100644
--- a/java/idea-ui/src/com/intellij/platform/templates/PlainModuleTemplatesFactory.java
+++ b/java/idea-ui/src/com/intellij/platform/templates/PlainModuleTemplatesFactory.java
@@ -54,7 +54,10 @@ public class PlainModuleTemplatesFactory implements ProjectTemplatesFactory {
   @Override
   public ProjectTemplate[] createTemplates(final String group, WizardContext context) {
     if (OTHER_GROUP.equals(group)) {
-      return context.isCreatingNewProject() ? new ProjectTemplate[] { new PlainModuleTemplate(new EmptyModuleBuilder() {
+      if (!context.isCreatingNewProject()) {
+        return ProjectTemplate.EMPTY_ARRAY;
+      }
+      return new ProjectTemplate[]{new PlainModuleTemplate(new EmptyModuleBuilder() {
         @Override
         public String getPresentableName() {
           return "Empty Project";
@@ -64,7 +67,7 @@ public class PlainModuleTemplatesFactory implements ProjectTemplatesFactory {
         public String getDescription() {
           return "Empty project without modules. Use it to create free-style module structure.";
         }
-      })} : ProjectTemplate.EMPTY_ARRAY;
+      })};
     }
     ModuleBuilder[] builders = context.getAllBuilders();
     return ContainerUtil.mapNotNull(builders, new NullableFunction<ModuleBuilder, ProjectTemplate>() {
